core/ArchiveProcessor/Loader.php

```diff
@@ -388,6 +389,43 @@ public function canSkipThisArchive()
         $isArchivingForcedWhenNoVisits = $this->shouldArchiveForSiteEvenWhenNoVisits();
         $hasSiteVisitsBetweenTimeframe = $this->hasSiteVisitsBetweenTimeframe($idSite, $params->getPeriod());
         $hasChildArchivesInPeriod = $this->dataAccessModel->hasChildArchivesInPeriod($idSite, $params->getPeriod());
+        $hasSegment = !$params->getSegment()->isEmpty();
+
+        if ($hasSegment) {
+            $periodEnd = $params->getPeriod()->getDateEnd();
+            $segmentHash = $params->getSegment()->getHash();
+            /** @var SegmentArchiving */
+            $segmentArchiving = StaticContainer::get(SegmentArchiving::class);
+            $segmentInfo = $segmentArchiving->findSegmentForHash($segmentHash, $idSite);
+
+            if ($segmentInfo) {
+                $segmentArchiveStartDate = $segmentArchiving->getReArchiveSegmentStartDate($segmentInfo);
+
+                if ($segmentArchiveStartDate !==null && $segmentArchiveStartDate->isLater($periodEnd->getEndOfDay())) {
+                    $doneFlag = Rules::getDoneStringFlagFor(
+                        [$idSite],
+                        $params->getSegment(),
+                        $params->getPeriod()->getLabel(),
+                        $params->getRequestedPlugin()
+                    );
+
+                    $table = Common::prefixTable('archive_invalidations');
+                    $sql = "SELECT idinvalidation FROM `$table` WHERE idsite = ? AND date1 = ? AND date2 = ? AND `period` = ? AND `name` = ?";
```

Comment:
This query could have a LIMIT 1 in case there are invalidations w/ the same params, but different `$report`s.

EDIT: actually, if we get a request to do a whole archive, but there is only an invalidation w/ a `report` value set, we should probably still skip. It might be needed to take into account any requested report (cc @tsteur).

Reply:
If we turn off the browser triggered archiving, when we load a report it will archive for those report, only for that one, am I right? Could it be happen somehow this gets into the invalidation table? Because if it can, we don't want to skip in this case, do we?

Reply:
> If we turn off the browser triggered archiving, when we load a report it will archive for that report, only for that one, am I right? 

I'm not sure what you mean. The 'archive only specific report' functionality is only triggered through core:archive and if there is a 'report' value set in archive_invalidations. This only happens if we reArchiveReport for a specific report. It never gets triggered via browser triggered archiving.

The issue I'm thinking of is if there is a request the archive, eg, `period = day, date = 2020-01-02, idSite = 2, segment = blahblahblah, plugin = Plugin` and the only item in the invalidation table is:

```
idsite = 2, period = day, date1 = 2020-01-02, date2 = 2020-01-02, name = done . md5(blahblahblah).Plugin, report = specificreport
```

In this case if everything else about the period/segment is true (it's outside of the process_new_segment_from range), we don't want to archive, because it would archive more then the specific report (it would do the whole plugin).

If we receive another request where report = specificreport as well, the we do want to archive because it's in the table.

Reply:
I see, I was confused. I thought about segment archiving in browser.

Reply:
I haven't modified the `reports` part but added the `limit 1`.

I'm going to add the specific report case and see how it goes with tests.